mlir/test/Dialect/Linalg/transform-op-decompose.mlir

```diff
@@ -211,31 +213,61 @@ func.func @softmax(%arg0: tensor<2x16x32xf32>, %dst: tensor<2x16x32xf32>) -> ten
 // CHECK-SAME:           %[[ARG0:[a-zA-Z0-9_]+]]: tensor<2x16x32xf32>, %[[DST:[a-zA-Z0-9_]+]]: tensor<2x16x32xf32>) -> tensor<2x16x32xf32> {
 // CHECK-DAG:        %[[D1:.+]] = tensor.empty() : tensor<2x16xf32>
 // CHECK-DAG:        %[[CST:.+]] = arith.constant -3.40282347E+38 : f32
-// CHECK:        %[[D2:.+]] = linalg.fill ins(%[[CST]] : f32) outs(%[[D1]] : tensor<2x16xf32>) -> tensor<2x16xf32>
+// CHECK:        %[[D2:.+]] = linalg.generic {indexing_maps = [#[[$MAP2]], #[[$MAP3]]],
```

Comment:
I'm wondering what `D` stands for in these tests 🤔 In any case, this would be a bit more descriptive.
```suggestion
// CHECK:        %[[FILL_1:.+]] = linalg.generic {indexing_maps = [#[[$MAP2]], #[[$MAP3]]],
```
[nit]

Reply:
I preserved the original naming for the diff to be readable. I also don't find the renaming very helpful or valuable.

Reply:
I am not sure why this becomes a generic op. Can we preserve the original behavior of this being a fill op?

Reply:
> I am not sure why this becomes a generic op. Can we preserve the original behavior of this being a fill op?

The reason is the decomposition now acts at linalg level and lives up to the promise of splitting a complex aggregate op into simpler ops at the same level of abstraction (btw, this is why it does make sense to have `LinalgOp`s in the decomposition result). After the decomposition, all the ops are non-generic linalg ops. At this stage, you have all the semantics available along with the benefit of no need to analyze affine maps (say, for fusion). After generalize you have a nice homogeneous set of generics as the last step in the test.

Reply:
> To me, more descriptive LIT variables are helpful - they make parsing tests much easier.

Renamed